test/units/modules/test_iptables.py

```diff
@@ -1,1169 +1,1299 @@
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertTrue(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 2)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
-        self.assertEqual(run_command.call_args_list[1][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-P',
-            'INPUT',
-            'ACCEPT',
-        ])
-
-    def test_policy_table_no_change(self):
-        """Test don't change policy of a chain if the policy is right"""
-        set_module_args({
-            'policy': 'ACCEPT',
-            'chain': 'INPUT',
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy ACCEPT)\n', ''),
-            (0, '', '')
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertFalse(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
-
-    def test_policy_table_changed_false(self):
-        """Test flush without parameters and change == false"""
-        set_module_args({
-            'policy': 'ACCEPT',
-            'chain': 'INPUT',
-            '_ansible_check_mode': True,
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy DROP)\n', ''),
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertTrue(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
+    return IPTABLES_CMD
+
+
+@pytest.fixture
+def mock_basic_commands(mocker):
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.get_bin_path",
+        return_value=get_bin_path(),
+    )
+    mocker.patch("ansible.modules.iptables.get_iptables_version", return_value=IPTABLES_VERSION)
+
+
+def test_without_required_parameters():
+    with pytest.raises(SystemExit) as e:
+        set_module_args({})
+        iptables.main()
+
+
+def test_flush_table_without_chain(mocker, mock_basic_commands):
+    """Test flush without chain, flush the table"""
```

Comment:
Typically, PEP 257 docstrings should be full sentences with a trailing period:
```suggestion
    """Test flush without chain, flush the table."""
```